admin/worker/run_autoscaler.go

```diff
@@ -18,10 +20,29 @@ func (w *Worker) runAutoscaler(ctx context.Context) error {
 	}
 
 	for _, rec := range recs {
-		// TODO: Add autoscaling logic based on the recommendation here.
-		// Consider checking rec.UpdatedOn to avoid making autoscaling decision on stale data!
-		w.logger.Info("autoscaler recommendation", zap.String("project_id", rec.ProjectID), zap.Int("recommended_slots", rec.RecommendedSlots))
-		break
+		duration := time.Since(rec.UpdatedOn)
+		if duration < 24*time.Hour {
+			w.logger.Debug("skipping autoscaler: the project has been scaled recently", zap.String("project_id", rec.ProjectID), zap.Time("project_updated_on", rec.UpdatedOn))
+			break
+		}
```

Comment:
Oh, okay, thx for clarifying!

Updating it to:
- check if the update of recommendation was over **7 days**, then skip
- since the autoscaler will be scheduled for a fixed time every week, I'm not going to add logic here for _checking when the slots were most recently updated_.

@begelundmuller  How would you like that?

Reply:
Makes sense, but I think we should reduce to 1 day (to be conservative and not scale in case of broken data pipelines)

Reply:
Agree.

(Resolving this conversation as it is followed up by https://github.com/rilldata/rill/pull/4376#discussion_r1538978689)